3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js

```
@@ -18,28 +18,36 @@ Full description at: https://github.com/HackYourFuture/Homework/blob/main/3-Usin
    should result in a network (DNS) error.
 ------------------------------------------------------------------------------*/
 function requestData(url) {
-  // TODO return a promise using `fetch()`
+  const response = fetch(url);
+  return response
+    .then((data) => data.json())
+    .catch((error) => {
+      console.log(error);
```

You should not catch the error here. Let it "bubble" up the call chain to the place where you expect to use the data. In this case that is the function `main()`. By handling the error here, the `catch` handler of `main()` never gets to see the error. Instead it tries to render non-existing data, which results in an JavaScript runtime error message being rendered in the page:

>TypeError: Cannot read properties of undefined (reading 'img')

I would have expected a message such as 'Fetch failed.'

Did you actually try point 6 of the instructions?

> 6. Test error handling, for instance, by temporarily changing the `.sh` in the url with `.shx`. There is no server at the modified url, therefore this should result in a network (DNS) error.

---

I have done all the changes in this exercise ..but I keep getting this error: 

> TypeError: Cannot read properties of undefined (reading 'img')

would you please check the commit?

---

Your `requestData()` function is not returning `data`. It just console.logs it. Therefore `requestData()` returns `undefined` and the `renderImage()` function chokes when it tries to access properties from the `data` object (which is not an object but `undefined`).

Also, you not use try/catch inside `requesData()` but instead let the error bubble up to `main()` where it is caught and handled.

---

I did return `data` and it worked fine.

I already push it